app/models/custom_attributes/custom_attribute_definition.rb

```diff
@@ -71,6 +77,37 @@ def update_custom_attribute(custom_attribute_params, selected_option_label)
       def default_option
         custom_attribute_options.find_by(id: default_value)
       end
+
+      def load_resoruce
+        respond_to?(:resource_type) ? resource_type : self.class.to_s.underscore.downcase.gsub!('_custom_attribute_definition', '')
+      end
+
+      def recurring_custom_attribute_previous
+        @recurring_custom_attribute_was = try(:is_recurring_was)
+      end
+
+      def update_recurring_custom_attribute
+        resource                     = load_resoruce
+        custom_attrubute_value_class = Object.const_get "#{resource.humanize}CustomAttributeValue"
+        custom_attribute_values      = custom_attrubute_value_class.where("#{resource}_custom_attribute_definition_id" => id)
+
+        if date_type? && !@recurring_custom_attribute_was.nil? && custom_attribute_values.present? && @recurring_custom_attribute_was != is_recurring
+          if @recurring_custom_attribute_was && !is_recurring
+            custom_attribute_values.update_all(recurring_date_value: nil)
+          else
+            custom_attribute_values_array = custom_attribute_values.map do |custom_attr_val|
+              [custom_attr_val.id, { recurring_date_value: custom_attr_val.date_for_next_alert }] if custom_attr_val.date_time_value.present?
+            end
+            custom_attribute_values_hash = custom_attribute_values_array.compact.to_h
+            custom_attrubute_value_class.update(custom_attribute_values_hash.keys, custom_attribute_values_hash.values)
```

Comment:
is this bulk update?
even if bulk update then think of a scenario where there are 20,000/ assets with the recurring service alert custom field?